server/config/config.go

```diff
@@ -34,37 +34,38 @@ type generalConfig struct {
+	LogIncludeShortFileName   bool               `yaml:"log_include_short_file_name" usage:"If true, log messages will include shortened originating file name."`
+	NoDefaultUserGroup        bool               `yaml:"no_default_user_group" usage:"Cloud-Only"`
+	LogEnableGCPLoggingFormat bool               `yaml:"log_enable_gcp_logging_format" usage:"If true, the output structured logs will be compatible with format expected by GCP Logging."`
+	LogErrorStackTraces       bool               `yaml:"log_error_stack_traces" usage:"If true, stack traces will be printed for errors that have them."`
+	TraceProjectID            string             `yaml:"trace_project_id" usage:"Optional GCP project ID to export traces to. If not specified, determined from default credentials or metadata server if running on GCP."`
+	TraceJaegerCollector      string             `yaml:"trace_jaeger_collector" usage:"Address of the Jager collector endpoint where traces will be sent."`
+	TraceServiceName          string             `yaml:"trace_service_name" usage:"Name of the service to associate with traces."`
+	TraceFraction             float64            `yaml:"trace_fraction" usage:"Fraction of requests to sample for tracing."`
+	TraceFractionOverrides    []string           `yaml:"trace_fraction_overrides" usage:"Tracing fraction override based on name in format name=fraction."`
+	IgnoreForcedTracingHeader bool               `yaml:"ignore_forced_tracing_header" usage:"If set, we will not honor the forced tracing header."`
+	CodeEditorEnabled         bool               `yaml:"code_editor_enabled" usage:"If set, code editor functionality will be enabled."`
+	UserManagementEnabled     bool               `yaml:"user_management_enabled" usage:"If set, the user management page will be enabled in the UI."`
+	GlobalFilterEnabled       bool               `yaml:"global_filter_enabled" usage:"If set, the global filter will be enabled in the UI."`
+	TestGridV2Enabled         bool               `yaml:"test_grid_v2_enabled" usage:"Whether to enable test grid V2"`
+	UsageEnabled              bool               `yaml:"usage_enabled" usage:"If set, the usage page will be enabled in the UI."`
+	UsageStartDate            string             `yaml:"usage_start_date" usage:"If set, usage data will only be viewable on or after this timestamp. Specified in RFC3339 format, like 2021-10-01T00:00:00Z"`
+	UsageTrackingEnabled      bool               `yaml:"usage_tracking_enabled" usage:"If set, enable usage data collection."`
+	DefaultRedisTarget        string             `yaml:"default_redis_target" usage:"A Redis target for storing remote shared state. To ease migration, the redis target from the remote execution config will be used if this value is not specified."`
+	DefaultShardedRedis       ShardedRedisConfig `yaml:"default_sharded_redis" usage:"Configuration for storing ephemeral state across multiple Redis instances. Mutually exclusive with default_redis_target."`
```

Comment:
nit: wdyt about calling this `default_redis_config`? (same for other config fields)

Reply:
I think we should keep the current name to make it clear that this is for setting up sharding.
I expect almost every on-prem customer to keep running BB using a non-sharded Redis setup and hopefully this naming will make it clearer that they shouldn't touch the sharded config.